cli/src/deno.rs

```diff
@@ -81,9 +84,29 @@ pub async fn run(extension_state: ExtensionState, entry_point: &str) -> Result<(
 }
 
 /// See https://github.com/denoland/deno/blob/main/core/examples/ts_module_loader.rs.
-struct TypescriptModuleLoader;
+struct PhylumExtensionsModuleLoader;
+
+impl PhylumExtensionsModuleLoader {
+    async fn load_from_filesystem(path: &Path) -> Result<String> {
+        let extensions_path = extensions_path()?;
+        if !path.starts_with(&extensions_path) {
+            return Err(anyhow!(
+                "{}: can't import modules outside of the extensions path {}",
+                path.to_string_lossy(),
+                extensions_path.to_string_lossy()
+            ));
+        }
+
+        Ok(fs::read_to_string(path).await?)
+    }
 
-impl ModuleLoader for TypescriptModuleLoader {
+    async fn load_from_deno_std(path: Url) -> Result<String> {
+        let response = reqwest::get(path).await?;
+        Ok(response.text().await?)
+    }
```

Comment:
The function makes it sound like this could only load from deno std, but it seems like there's no restriction on the remote?

Reply:
Agreed, same as with the related comment below.